test/offsetarray.jl

```diff
@@ -386,11 +386,6 @@ I,J,N = findnz(z)
 v  = OffsetArray([1,1e100,1,-1e100], (-3,))*1000
 v2 = OffsetArray([1,-1e100,1,1e100], (5,))*1000
 @test isa(v, OffsetArray)
-cv  = OffsetArray([1,1e100,1e100,2], (-3,))*1000
-cv2 = OffsetArray([1,-1e100,-1e100,2], (5,))*1000
-@test isequal(cumsum_kbn(v), cv)
-@test isequal(cumsum_kbn(v2), cv2)
-@test isequal(sum_kbn(v), sum_kbn(parent(v)))
```

Comment:
Perhaps modify rather than eradicate these tests? I conjecture that testing `OffsetArray` is these tests' purpose rather than testing `[cum]sum_kbn` specifically :).

Reply:
Interestingly, even if I convert these to `BigFloat`s, using `cumsum` rather than `cumsum_kbn` seems to give a different answer:
```
julia> v = OffsetArray(BigFloat[1,1e100,1,-1e100], (-3,))*1000
OffsetArray{BigFloat,1,Array{BigFloat,1}} with indices -2:1:
  1.0e+03
  1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+103
  1.0e+03
 -1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+103

julia> cumsum(v)
OffsetArray{BigFloat,1,Array{BigFloat,1}} with indices -2:1:
 1.0e+03
 1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+103
 1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+103
 1.0e+03

julia> cumsum_kbn(v)
OffsetArray{BigFloat,1,Array{BigFloat,1}} with indices -2:1:
 1.0e+03
 1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+103
 1.000000000000000015902891109759918046836080856394528138978132755774783877217038e+103
 2.0e+03
```

Reply:
Perhaps need higher precision, and/or `cumsum` doesn't preserve precision in the accumulation?

Reply:
I'd just use `isapprox` here. The fact that the result was exact before must have been due to the way the KBN algorithm worked in that particular case, but we don't really care here since the goal is just to test that it works on `OffsetArray`.

Reply:
The problem with `isapprox` in this case is that 1000 and 2000 aren't approximately equal. 😛

Reply:
OK, I misread. That's just because `BigFloat` doesn't have enough precision by default. It works with a higher precision:
```julia
julia> setprecision(BigFloat, 500)
500

julia> cumsum(BigFloat[1,1e100,1,-1e100])
4-element Array{BigFloat,1}:
 1.0                                                                                                        
 1.0000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815105e+100
 1.0000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815106e+100
 2.0                                                                                                        

julia> cumsum_kbn(BigFloat[1,1e100,1,-1e100])
4-element Array{BigFloat,1}:
 1.0                                                                                                        
 1.0000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815105e+100
 1.0000000000000000159028911097599180468360808563945281389781327557747838772170381060813469985856815106e+100
 2.0                                               
```

But it's really weird to mix testing `OffsetArray` with testing precision with very high values. Maybe split them into two separate tests, with a comment saying what is being tested?